### ISSUE TITLE
Replica failing with `ERROR:  schema "^B" does not exist`              

### ISSUE THREAD

**Issue by michelp** [2ndQuadrant/pglogical#267]
Hello,

We are using pglogical to replicate data from the real-time server in RDS to a warehouse replica.  The produce is 12.2 running pglogical 2.3.0 (we cannot control this), and the consumer is pg 12.3 running pglogical 2.3.1.   It's been humming along nicely now for a couple of weeks, replicating about 10-50M rows a day.  Today it started throwing similar but slightly different errors on each replica, this one subscription1:

```2020-07-06 00:20:08.452 UTC [7615] [unknown]@prod LOG:  starting apply for subscription subscription1          
2020-07-06 00:20:09.075 UTC [7615] [unknown]@prod ERROR:  schema "^Bublic" does not exist                      
2020-07-06 00:20:09.075 UTC [7615] [unknown]@prod CONTEXT:  apply INSERT in commit before 9E9/B86A7370, xid 1331216 committed at 2020-07-06 00:00:50.203562+00 (action #3500) from node replorigin 1                           
2020-07-06 00:20:09.075 UTC [7615] [unknown]@prod LOG:  apply worker [7615] at slot 1 generation 630 exiting with error
```

and this on subscription 2:

```
2020-07-06 00:24:10.862 UTC [8642] [unknown]@prod ERROR:  schema "^B" does not exist                           
2020-07-06 00:24:10.862 UTC [8642] [unknown]@prod CONTEXT:  apply INSERT in commit before 9EA/19EDCF88, xid 1331783 committed at 2020-07-06 00:18:04.733991+00 (action #3854) from node replorigin 1                           
2020-07-06 00:24:10.862 UTC [8642] [unknown]@prod LOG:  apply worker [8642] at slot 1 generation 1856 exiting with error
2020-07-06 00:24:10.868 UTC [21965] LOG:  background worker "pglogical apply 16384:1871150101" (PID 8642) exited with exit code 1
```

The producer log contains a lot of these which I think are the replica connection attempts:

```
2020-07-06 00:09:58 UTC:10.192.3.20(42690):icarusdbuser@dev:[7661]:LOG: starting logical decoding for slot "pgl_prod_icarus_provider_subscription1"
2020-07-06 00:09:58 UTC:10.192.3.20(42690):icarusdbuser@dev:[7661]:DETAIL: Streaming transactions committing after 9E9/B869BF90, reading WAL from 9E9/B1900470.
2020-07-06 00:09:58 UTC:10.192.3.20(42690):icarusdbuser@dev:[7661]:LOG: logical decoding found consistent point at 9E9/B1900470
2020-07-06 00:09:58 UTC:10.192.3.20(42690):icarusdbuser@dev:[7661]:DETAIL: Logical decoding will begin using saved snapshot.
2020-07-06 00:09:58 UTC:10.192.3.20(42690):icarusdbuser@dev:[7661]:LOG: could not send data to client: Connection reset by peer
2020-07-06 00:09:58 UTC:10.192.3.20(42690):icarusdbuser@dev:[7661]:CONTEXT: slot "pgl_prod_icarus_provider_subscription1", output plugin "pglogical_output", in the change callback, associated LSN 9E9/B7E410A8
```

I don't know how this control B character could be introduced, all scripts are written in normal text and a literal search for that byte in our source found nothing.    I've tried dropping and readding the subscriptions and not sure what to do next to debug this issue.

Please let me know if there's any additional information I can provide, the replication is currently stopped, and short of doing a full rebuild I'd like to understand the issue more before doing that.

EDIT

I should add that a procedure runs at midnight that creates new partitions, adds them to the repset, this procedure has been running for several weeks now just fine, so not sure if that's relevant but here it is.

```
select pglogical.replicate_ddl_command($ddl$
CREATE OR REPLACE PROCEDURE public.partition2(conf jsonb, parent_name text) LANGUAGE plpgsql
AS $$
DECLARE
    start_time timestamptz;
    end_time timestamptz;
    new_name text;
    partition_interval interval;
BEGIN
    partition_interval = format('%s %s', conf->>'INTERVAL', conf->>'UNIT')::interval;
    start_time = date_trunc(conf->>'UNIT', now());
    end_time = start_time + partition_interval;
    new_name = parent_name || '_' || replace(start_time::text || '-' || end_time::text, ' ', '-');

    IF EXISTS
        (SELECT * FROM partition_of WHERE
        inhparent = (SELECT oid FROM pg_class
             WHERE relkind = 'p' AND relname = parent_name)
        AND name = new_name AND relkind = 'r')
    THEN
       RAISE NOTICE '% already exists', new_name;
       RETURN;
    END IF;

    EXECUTE format('LOCK TABLE %I IN SHARE MODE', parent_name);

    RAISE NOTICE 'Creating partition %', new_name;

    -- create new partition table, not attached yet
    PERFORM pglogical.replicate_ddl_command(format(
    'CREATE TABLE IF NOT EXISTS public.%I (LIKE public.%I INCLUDING DEFAULTS INCLUDING CONSTRAINTS)',
    new_name, parent_name));

    PERFORM pglogical.replicate_ddl_command(format(
    'COMMENT ON TABLE public.%I IS %L', new_name,
    format('%s partition from %s to %s', parent_name, start_time, end_time)));

    PERFORM pglogical.replication_set_add_table('default_insert_only', new_name);

    -- move any rows in ping_default that should be in this new table
    EXECUTE format(
    'WITH moved_rows AS (
        DELETE FROM %I_default a -- WHERE seen <@ $1
        RETURNING a.*)
     INSERT INTO %I SELECT * FROM moved_rows;',
    parent_name, new_name)
    USING tstzrange(start_time, end_time);

    -- attach new partition
    PERFORM pglogical.replicate_ddl_command(format(
    'ALTER TABLE public.%I ATTACH PARTITION public.%I FOR VALUES FROM (%L) TO (%L)',
    parent_name, new_name,
    start_time,
    end_time));
END;
$$;
$ddl$);
```

Thanks!

**Comment by michelp** [comment]
I was able to recover the situation by dropping the nodes on the producer and consumers and reinitializing them.  We can backfill the missing data.  Not sure if this is a pglogical bug, but everything seems fine now.  If you want me to keep this open in the event you'd like to investigate otherwise I'll close it.

**Comment by martinmarques** [comment]
2.3.0 shouldn't be used anywhere. That's why we released 2.3.1 (and 2.3.2 later). 
If AWS is still providing that in RDS you should bring it up to them.

**Comment by michelp** [comment]
Understood, it looks like between the time we checked and now they have released pg 12.3 with pglogical 2.3.1.  We'll upgrade asap.
